screens/SCWTest.tsx

```diff
@@ -0,0 +1,165 @@
+import { base } from "thirdweb/chains";
+import {
+  useActiveAccount,
+  useActiveWallet,
+  useConnect,
+  useDisconnect,
+  useSendTransaction,
+  useSetActiveWallet,
+  useSwitchActiveWalletChain,
+} from "thirdweb/react";
+
+import config from "../config";
+
+const MainScreen: React.FC = () => {
+  const { connect: thirdwebConnect } = useConnect();
+  const activeAccount = useActiveAccount();
+  const activeWallet = useActiveWallet();
+  const { disconnect: thirdwebDisconnect } = useDisconnect();
+  const sendTransaction = useSendTransaction();
+  const setActiveWallet = useSetActiveWallet();
+  const switchChain = useSwitchActiveWalletChain();
+
+  const disconnectWithoutLogin = useCallback(async () => {
+    const storageKeys = await AsyncStorage.getAllKeys();
+    const wcKeys = storageKeys.filter((k) =>
+      k.startsWith("-Coinbase Smart Wallet:")
+    );
+    await AsyncStorage.multiRemove(wcKeys);
+
+    console.log("DONE!");
+  }, []);
+
+  const handleButtonPress = useCallback(() => {
+    thirdwebConnect(async () => {
+      // instantiate wallet
+      const coinbaseWallet = thirdwebWallets["Coinbase Smart Wallet"];
+      let connected = false;
+      try {
+        console.log("auto-connecting");
+        await coinbaseWallet.autoConnect({ client: thirdwebClient });
+        connected = true;
+        console.log("auto-connected");
+        setActiveWallet(coinbaseWallet);
+      } catch (e) {
+        console.log(e);
+      }
+      if (connected) return coinbaseWallet;
+      try {
+        console.log("connecting");
+        await coinbaseWallet.connect({ client: thirdwebClient });
+        console.log("connected");
+        setActiveWallet(coinbaseWallet);
+      } catch (e) {
+        console.log(e);
+      }
+      return coinbaseWallet;
+    });
+  }, [thirdwebConnect, setActiveWallet]);
+
+  const handleTransaction = useCallback(async () => {
+    console.log("clicked!");
+    const transaction = prepareTransaction({
+      to: "0x2376e9C7C604D1827bA9aCb1293Dc8b4DA2f0DB3",
+      value: BigInt(10000000000000),
+      chain: base,
+      client: thirdwebClient,
+    });
+    console.log("transaction", transaction);
+    try {
+      await switchChain(base);
+      console.log("switched chain");
+      const tx = await sendTransaction.mutateAsync(transaction);
+      console.log("sent transaction", tx.transactionHash);
+    } catch (e: any) {
+      alert(e.message);
+      console.log(e);
+    }
+  }, [switchChain, sendTransaction]);
+
+  const handleSignature = useCallback(async () => {
+    if (!activeAccount) return;
+    await switchChain(base);
+    const signer = await ethers5Adapter.signer.toEthers({
+      client: thirdwebClient,
+      chain: base,
+      account: activeAccount,
+    });
+    const signature = await signer.signMessage("Hello, world");
+    console.log("base", signature);
+  }, [activeAccount, switchChain]);
+
+  return (
+    <View style={styles.container}>
+      {!activeAccount && (
```

Comment:
if we're going to use this jsx, can you extract into more bite sized components?

Reply:
file removed